src/lib/comms/sol-http-server-impl-microhttpd.c

```diff
@@ -960,6 +1038,80 @@ sol_http_server_send_response(struct sol_http_request *request, struct sol_http_
     return ret;
 }
 
+SOL_API struct sol_http_response_progressive *
+sol_http_server_send_response_progressive(struct sol_http_request *request,
+    const struct sol_http_response *response,
+    void (*on_del)(void *data, const struct sol_http_response_progressive *progressive),
+    const void *cb_data)
+{
+    int ret;
+    struct sol_http_response_progressive *progressive;
+    struct MHD_Response *mhd_response;
+
+    SOL_NULL_CHECK(request, NULL);
+    SOL_NULL_CHECK(request->connection, NULL);
+    SOL_NULL_CHECK(response, NULL);
+
+    progressive = calloc(1, sizeof(*progressive));
+    SOL_NULL_CHECK(progressive, NULL);
+
+    sol_buffer_init(&progressive->buffer);
+    progressive->connection = request->connection;
+    progressive->delete_me = false;
+
+    MHD_resume_connection(request->connection);
+
+    mhd_response = build_mhd_response_progressive(response, progressive);
+    SOL_NULL_CHECK_GOTO(mhd_response, err);
+
+    ret = MHD_queue_response(request->connection, SOL_HTTP_STATUS_OK, mhd_response);
```

Comment:
shouldn't we get the status from the `response`? This may be used to deliver everything, not just successful content.